fabtests/common/shared.c

```diff
@@ -2397,6 +2414,14 @@ static int ft_fdwait_for_comp(struct fid_cq *cq, uint64_t *cur,
 			if (!ft_tag_is_valid(cq, &comp, ft_tag ? ft_tag : rx_cq_cntr))
 				return -FI_EOTHER;
 			(*cur)++;
+
+			if (comp.flags & FI_REMOTE_CQ_DATA) {
+				rx_cq_entry_has_data = true;
+				rx_cq_data = comp.data;
+			} else {
+				rx_cq_entry_has_data = false;
+			}
+
```

Comment:
We have a fabtest that verifies cq_data.  Sure, it's lame, but is this much of an improvement?  Second, the above structure is repeated 3 times.  Why not just check the cq data directly in all places, similar to how the tag check is done?  Saving the information to check later is going to make it hard to figure out what's happening and make changes more difficult.

Finally, the update to the man page was to define 'len' as valid for RMA writes with cq data.  The request was for something in fabtests to verify that len is actually set, so we can identify what providers need to be updated to support the new definition.

Reply:
> We have a fabtest that verifies cq_data. Sure, it's lame, but is this much of an improvement? Second, the above structure is repeated 3 times. Why not just check the cq data directly in all places, similar to how the tag check is done? Saving the information to check later is going to make it hard to figure out what's happening and make changes more difficult.

I just noticed that there is `fi_cq_data` test, that test senddata, I am working on adding `writedata` support to it.

> Finally, the update to the man page was to define 'len' as valid for RMA writes with cq data. The request was for something in fabtests to verify that len is actually set, so we can identify what providers need to be updated to support the new definition.

So we can merge the 1st commit as is, and I will open a separate PR to add the test.

Does the plan sounds right?

Reply:
ubertest already supports writedata, so I don't know that we even need a new test for cq data.

The problem with the man page update is that it is defining new API requirements, and I don't think that any of the providers actually support that requirement.  We need to coordinate updating all of the providers with the man page.

What we don't have is a fabtest that verifies the len field in the cq_entry is set correctly for an RMA write with cq data.  We're not trying to validate the cq data, but the len.  And I expect that if we add that test, CI will fail because the providers aren't taking that action.

Reply:
I can add verfication of `len` to `fi_cq_data`, and open PR.

The PR will trigger AWS CI, which will test shm, tcp, sockets and EFA.

Reply:
You can also update ubertest.  There's a ft_verify_comp() call for this purpose.